ambry-cloud/src/main/java/com.github.ambry.cloud/azure/AzureCloudDestination.java

```diff
@@ -128,25 +130,36 @@ public boolean uploadBlob(BlobId blobId, long blobSize, CloudBlobMetadata cloudB
 
     BlobRequestOptions options = null; // may want to set BlobEncryptionPolicy here
     OperationContext opContext = null;
+    azureMetrics.blobUploadRequestCount.inc();
     try {
+      Timer.Context storageTimer = azureMetrics.blobUploadTime.time();
       CloudBlobContainer azureContainer = getContainer(blobId, true);
-      CloudBlockBlob azureBlob = azureContainer.getBlockBlobReference(blobId.getID());
+      String azureBlobName = getAzureBlobName(blobId);
+      CloudBlockBlob azureBlob = azureContainer.getBlockBlobReference(azureBlobName);
 
       if (azureBlob.exists()) {
         logger.debug("Skipping upload of blob {} as it already exists in Azure container {}.", blobId,
             azureContainer.getName());
+        azureMetrics.blobSkippedCount.inc();
```

Comment:
Is it better to unify prefix of `blobSkippedCount `, `blobUploadedCount ` and `blobUploadErrorCount `?

`blobUploadSkipCount`, `blobUploadSuccessCount` and `blobUploadErrorCount`

Reply:
Changed to blobUploadSkippedCount